packages/pyright-internal/src/common/pythonVersion.ts

```diff
@@ -118,38 +130,38 @@ export namespace PythonVersion {
         return false;
     }
 
-    export function isGreaterOrEqualTo(version: PythonVersion, other: PythonVersion) {
+    export function isGreaterOrEqualTo(version: PythonVersion | undefined, other: PythonVersion | undefined) {
         return isEqualTo(version, other) || isGreaterThan(version, other);
     }
 
-    export function isLessThan(version: PythonVersion, other: PythonVersion) {
+    export function isLessThan(version: PythonVersion | undefined, other: PythonVersion | undefined) {
         return !isGreaterOrEqualTo(version, other);
     }
 
-    export function isLessOrEqualTo(version: PythonVersion, other: PythonVersion) {
+    export function isLessOrEqualTo(version: PythonVersion | undefined, other: PythonVersion | undefined) {
         return !isGreaterThan(version, other);
     }
 
-    export function toMajorMinorString(version: PythonVersion): string {
-        return `${version.major}.${version.minor}`;
+    export function toMajorMinorString(version: PythonVersion | undefined): string {
+        return `${version?.major}.${version?.minor}`;
```

Comment:
I don't think we should be allowed to print a non-existent version. Or if we do, it shouldn't appear as `undefined.undefined`.

Reply:
Good point. I'll make it print an empty string.